sources/packages/backend/apps/api/src/services/disbursement-schedule/disbursement-schedule-service.ts

```diff
@@ -39,56 +42,15 @@ export class DisbursementScheduleService extends RecordDataModelService<Disburse
     enrollmentPeriod: EnrollmentPeriod,
     paginationOptions: PaginationOptions,
   ): Promise<PaginatedResults<DisbursementSchedule>> {
-    const coeThresholdDate = addDays(COE_WINDOW);
-    const coeQuery = this.repo
-      .createQueryBuilder("disbursementSchedule")
-      .select([
-        "disbursementSchedule.id",
-        "disbursementSchedule.disbursementDate",
-        "disbursementSchedule.coeStatus",
-        "application.applicationNumber",
-        "application.id",
-        "studentAssessment.id",
-        "currentAssessment.id",
-        "offering.studyStartDate",
-        "offering.studyEndDate",
-        "student.id",
-        "user.firstName",
-        "user.lastName",
-      ])
-      .innerJoin("disbursementSchedule.studentAssessment", "studentAssessment")
-      .innerJoin("studentAssessment.application", "application")
-      .innerJoin("application.currentAssessment", "currentAssessment")
-      .innerJoin("currentAssessment.offering", "offering")
-      .innerJoin("offering.institutionLocation", "location")
-      .innerJoin("application.student", "student")
-      .innerJoin("student.user", "user")
-      .where("studentAssessment.id = currentAssessment.id")
-      .andWhere("location.id = :locationId", { locationId })
-      .andWhere("application.applicationStatus IN (:...status)", {
-        status: [ApplicationStatus.Enrolment, ApplicationStatus.Completed],
-      })
-      .andWhere("disbursementSchedule.hasEstimatedAwards = true");
-    if (enrollmentPeriod === EnrollmentPeriod.Upcoming) {
-      coeQuery.andWhere(
-        new Brackets((qb) => {
-          qb.where(
-            "disbursementSchedule.disbursementDate > :coeThresholdDate",
-          ).orWhere("disbursementSchedule.coeStatus != :required", {
-            required: COEStatus.required,
-          });
-        }),
+    const disbursementCOEQuery =
+      this.confirmationOfEnrollmentService.getDisbursementForCOEQuery(
+        this.repo,
+        enrollmentPeriod === EnrollmentPeriod.Current,
```

Comment:
IMO, using the `EnrollmentPeriod` instead of a `bool` looks easier to read the code. 
The options `EnrollmentPeriod` can be considered a concept beyond the API scope.